tests/elements-regression/tests/regression_with_templates.test.js

```diff
@@ -0,0 +1,37 @@
+import { expect, test } from '@playwright/test';
+import _path from 'path';
+import WpAdminPage from '../src/pages/wp-admin-page';
+import EditorPage from '../src/pages/editor-page';
+
+test.describe( 'Elementor regression tests with templates for CORE', () => {
+	test.beforeAll( async ( { browser } ) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+		const wpAdmin = new WpAdminPage( page );
+		await wpAdmin.setExperiments( {
+			container: 'active',
+		} );
+	} );
+
+	const testData = [ 'divider', 'heading', 'text_editor' ];
+	for ( const widgetType of testData ) {
+		test( `Test ${ widgetType } template`, async ( { page } ) => {
+			const filePath = _path.resolve( __dirname, `./templates/${ widgetType }.json` );
+			const wpAdminPage = new WpAdminPage( page );
+			const editorPage = new EditorPage( page );
+
+			await editorPage.setupElementorPage( wpAdminPage );
+			await editorPage.loadTemplate( filePath );
+			await editorPage.setElementorEditorPanel( 'Preview' );
+			const widgetCount = await editorPage.getWidgetCount();
+			for ( let i = 0; i < widgetCount; i++ ) {
+				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
+			}
+			await editorPage.setElementorEditorPanel( 'Back to Editor' );
+			await editorPage.publishAndViewPage();
+			await page.waitForTimeout( 500 );
```

Comment:
Do we need this?

Reply:
removed from here, but before publish we have to use explicit timeout. I didn't found a solution of how to wait until all widgets are rendered. It's hard to do it in Iframe. Event this didn't help: `await editorPage.waitForElementRender( widgetIds[ 0 ] );`